modules/project-common.nix

```diff
@@ -24,6 +24,31 @@ with lib.types;
           shell = { tools.cabal = {}; crossPlatforms = p: [ p.ghcjs ]; }
       '';
     };
+    evalSystem = mkOption {
+      type = str;
+      default = builtins.currentSystem or pkgs.buildPackages.system;
+      description = ''
+        Specifies the system on which `cabal configure` and `plan-to-nix` should run.
+        If not specified the `evalPackages` default will be used (builtins.currentSystem or
+        the `buildPackages` system).
+        This argument is useful when using pure flake evalution (where builtins.currentSystem does
+        not exist) when there are no builders present for the some or all of the flakes supported
+        systems (so `buildPackages` also fails).
+      '';
+    };
+    evalPackages = mkOption {
+      type = attrs;
+      default =
+        if pkgs.system == config.evalSystem
+          then pkgs
+        else if pkgs.buildPackages.system == config.evalSystem
+          then pkgs.buildPackages
+        else
+          import pkgs.path {
```

Comment:
could also throw in this case and require the user to specify?

Reply:
I've changed the default to use `buildPackages`.  I also removed the default to make sure nothing in `haskell.nix` itself (including the CI) depends on the default.  The `import` code here should now only get used if `evalSystem` was explicitly passed in (for instance if the user passed `evalSystem = builtins.currentSystem;`).